mule-extensions-api/src/main/java/org/mule/runtime/extension/api/annotation/execution/OnTerminate.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.annotation.execution;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import org.mule.runtime.extension.api.OnTerminateResult;
+import org.mule.runtime.extension.api.runtime.source.Source;
+import org.mule.runtime.extension.api.runtime.source.SourceCallbackContext;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Used to indicate that an annotated method should be executed when a given
+ * execution finishes.
+ *
+ * The actual meaning of the term 'given execution' depends of the context in which
```

Comment:
Where other than a source can this be used? If the answer is `nowhere` then just speak about sources.